### ISSUE TITLE
Implementation of Show and Tell

### ISSUE THREAD

**Issue by wstmac** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#112]
If I donot include attention module in the model, can it be treat as the implementation of the model in the paper show and tell? And if I want to train the model to generate multi-style image caption, which model is better to be modified, is the show-attend-tell or show-and-tell?

**Comment by kmario23** [comment]
Hey @wstmac,
  In principle, yes. However it won't be the exact replica of S&T since that uses [GoogLeNet CNN with Batch Normalization](https://arxiv.org/pdf/1502.03167.pdf).

To do so, you can [simply comment out the doubly stochastic reg. after you compute the loss](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning/blob/master/train.py#L186). Additionally, you'd also want to add batchnorm since this codebase doesn't use it.

>  And if I want to train the model to generate multi-style image caption, which model is better to be modified, is the show-attend-tell or show-and-tell?

could you please elaborate on what you meant by "multi-style" image caption? And what dataset are you planning to use for this?

**Comment by wstmac** [comment]
Hi @kmario23 

These years there are many works focus on generate multi-style image caption like, [generating romantic/humorous sentences](https://www.microsoft.com/en-us/research/uploads/prod/2017/06/Generating-Attractive-Visual-Captions-with-Styles.pdf), [generate positive/negative image caption](http://openaccess.thecvf.com/content_CVPR_2019/papers/Guo_MSCap_Multi-Style_Image_Captioning_With_Unpaired_Stylized_Text_CVPR_2019_paper.pdf) etc. Some of the works are using the encoder-decoder method, and I found they do not use the attention module in their models. Is it because the attention module will hurt the performance of the final model? Let's suppose I have a dataset with not only image-caption pair, but also caption-style pair (positive/negative/romantic/humorous style), when I want to train a encoder-decoder model, do I need to include the attention module in the model?
